src/auro-button.js

```diff
@@ -28,12 +30,15 @@ import '@alaskaairux/auro-loader';
  * @attr {String} variant - Sets button variant option. Possible values are: `secondary`, `tertiary`
  * @prop {Boolean} ready - When false the component API should not be called.
  * @fires auroButton-ready - Notifies that the component has finished initializing.
+ * @fires auroButton-toggledText - Notifies that the visibility of the default slot content has changed.
  *
- * @slot - Provide text for the button.
+ * @slot - Default slot for the text of the button.
+ * @slot icon - Slot to provide auro-icon for the button.
  */
+
+/* eslint-disable max-statements, no-negated-condition */
```

Comment:
why did we disable these rules?

Reply:
The constructor has more statements than allowed by eslint and we use a negated condition to toggle the `hideText` attribute.

Reply:
Hmm the constructor rule breaking is hard to avoid. I'm not sure if auro has a pattern for ignoring these kinds of rules but it might make sense to [only ignore top level functions for now](https://eslint.org/docs/latest/rules/max-statements#ignoretoplevelfunctions)?

The negated conditional however appears to be in line 206 and easy to avoid if we use `this.hideText` as the condition instead of `!this.hideText`.
<img width="724" alt="image" src="https://github.com/AlaskaAirlines/auro-button/assets/23248722/567f0585-f051-4f72-979d-d83ba8b00bf3">